sound/soc/sof/intel/hda.c

```diff
@@ -759,6 +759,10 @@ int hda_dsp_remove(struct snd_sof_dev *sdev)
 	pci_free_irq_vectors(pci);
 
 	hda_dsp_stream_free(sdev);
+
+	iounmap(sdev->bar[HDA_DSP_BAR]);
+	iounmap(bus->remap_addr);
+
 	return 0;
```

Comment:
these are good, but I still like adding conditions such as 'if (bus->remap_addr)'.

Reply:
it's not necessary, you will not have a case where this is NULL since .remove is not called when .probe fails?